pdns/syncres.hh

```diff
@@ -339,7 +346,30 @@ public:
   };
 
   typedef std::unordered_map<DNSName, DecayingEwmaCollection> nsspeeds_t;
-  typedef map<ComboAddress, EDNSStatus> ednsstatus_t;
+  
+  struct EDNSStatus {
+    EDNSStatus(const ComboAddress &arg) : address(arg) {}
+    ComboAddress address;
+    time_t modeSetAt{0};
+    enum EDNSMode { UNKNOWN=0, EDNSOK=1, EDNSIGNORANT=2, NOEDNS=3 } mode{UNKNOWN};
+  };
+  
+  struct ednsstatus_t : public multi_index_container<EDNSStatus,
+                                indexed_by<
+                                  ordered_unique<member<EDNSStatus, ComboAddress, &EDNSStatus::address>>,
```

Comment:
I'd advise the use of named indexes, using tags, see in `recpacketcache.hh` for an example. It is much easier to know which index you are using later without having to look at the definition, and prevent renumbering if you ever want to add an index.

Reply:
Will do.